### DISCUSSION TITLE
librespot -> pipe

### DISCUSSION THREAD

**Discussion by ssaferga** [snapcast/snapcast#986] [Q&A]
Hi,

Would it be possible to directly output the librespot to a pipe/fifo?

I want to use my spotify to cast to the snapcast, then output it to a pipe so i can use the pipe for other services such as owntone. I'm able to use snapcast and mopidy to play to a pipe, and use the same pipe in owntone. But it would be much more convenient to just use the spotify client to snapcast and then get it on owntone (I have a mix of airplay and bluetooth receivers)

in the snapserver.conf something like;:
[stream]
source = librespot:///librespot?name=ssaferga&devicename=Snapcast&username=secret&password=secret&killall=false&bitrate=320&volume=100&params=--zeroconf-port%205353&params=--device-pipe?/path/to/pipelocation

**Comment by vjdw** [comment]
The `librespot:///` source tells snapserver to start an instance of librespot and read audio from its stdout.

Would it solve your problem to separately start librespot using its `--backend pipe` option and then configure it as a `pipe:///` source in snapserver?

Something like:
`~/.cargo/bin/librespot -n Home -b 320 --initial-volume 100 --backend pipe --device /tmp/snapfifospotify`
where `/tmp/snapfifospotify` is the path to a pipe you've created with `mkfifo`.

**Reply by ssaferga** [reply]
I will try to do that, Thanks for the quick response!
